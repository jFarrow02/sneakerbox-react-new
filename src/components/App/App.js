import React from 'react';
import {BrowserRouter} from 'react-router-dom';
import './App.scss';
import {
  Banner,
  Content,
  Footer,
  Nav,
} from '../index';

function App() {

  return (
    <BrowserRouter>
      <div className="App">
        <Banner title='SneakerBox'/>
        <Nav/>
        <Content/>
        <Footer/>
      </div>
    </BrowserRouter>
    
  );
}

export default App;
