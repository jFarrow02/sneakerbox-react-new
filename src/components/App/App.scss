@import '../../styles/variables.scss';

.App {
  display: grid;
  background-color: $copper;
  #row-1, #row-2, #row-3{
    grid-column: 1 /span 2;
  }
  #row-2{
    display: grid;
    grid-template-columns: repeat(12, 1fr);
  }
}
