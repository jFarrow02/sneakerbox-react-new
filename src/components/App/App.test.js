import React from 'react';
import ReactDOM from 'react-dom';
import App from './App';
import {configure, shallow} from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import renderer from 'react-test-renderer';

configure({adapter: new Adapter()});
const noProps = shallow(<App/>);

it('renders without crashing', () => {
  const div = document.createElement('div');
  ReactDOM.render(<App />, div);
  ReactDOM.unmountComponentAtNode(div);
});

it('should match a snapshot', ()=>{
  const tree = renderer.create(<App/>)
    .toJSON();
  expect(tree).toMatchSnapshot();
});

it('should render when passed no props', ()=>{
  expect(noProps.exists()).toEqual(true);
});
