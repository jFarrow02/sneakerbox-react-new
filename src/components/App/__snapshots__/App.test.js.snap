// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should match a snapshot 1`] = `
<div
  className="App"
>
  <header
    className="Banner"
  >
    <h1>
      SneakerBox
    </h1>
    <div
      className="Cart"
    >
      Cart goes here
      <h4>
        No Items
      </h4>
    </div>
  </header>
  <nav
    className="Nav"
  >
    <a
      href="/home"
      onClick={[Function]}
    >
      Home
    </a>
    <a
      href="/products/men"
      onClick={[Function]}
    >
      Men
    </a>
    <a
      href="/products/women"
      onClick={[Function]}
    >
      Women
    </a>
    <a
      href="/products/kids"
      onClick={[Function]}
    >
      Kids
    </a>
    <a
      href="/products/sale"
      onClick={[Function]}
    >
      Sale!
    </a>
    <a
      href="/myaccount"
      onClick={[Function]}
    >
      My Account
    </a>
  </nav>
  <main
    className="Content"
  >
    <section
      className="Home"
    >
      <div
        className="CaptionedImage"
      >
        <div
          className="CaptionedImage--caption"
          style={
            Object {
              "left": "15px",
              "position": "absolute",
              "top": "15px",
            }
          }
        >
          <div
            className="Home--caption"
          >
            <h2>
              Winter Mode
            </h2>
            <p>
              Get ready for rugged adventures with our new 
              <strong>
                Winter Collection.
              </strong>
            </p>
            <a
              href="#"
            >
              &gt;&gt;Shop Now.
            </a>
          </div>
        </div>
        <img
          alt="Main image home page; man walking along railway"
          height=""
          src="sneakers-railtracks.jpg"
          title=""
          width=""
        />
      </div>
    </section>
  </main>
  <footer
    className="Footer"
  >
    <div
      className="Footer--icon-row"
    >
      <a
        aria-label="Sneakerbox on Twitter"
        href="#"
        onClick={null}
        target="_self"
      >
        <svg
          alt="twitter"
          className="twitter-icon"
          focusable="false"
          height="50px"
          onMouseEnter={[Function]}
          onMouseLeave={[Function]}
          viewBox="0 -47 512.00203 512"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="m191.011719 419.042969c-22.140625 0-44.929688-1.792969-67.855469-5.386719-40.378906-6.335938-81.253906-27.457031-92.820312-33.78125l-30.335938-16.585938 32.84375-10.800781c35.902344-11.804687 57.742188-19.128906 84.777344-30.597656-27.070313-13.109375-47.933594-36.691406-57.976563-67.175781l-7.640625-23.195313 6.265625.957031c-5.941406-5.988281-10.632812-12.066406-14.269531-17.59375-12.933594-19.644531-19.78125-43.648437-18.324219-64.21875l1.4375-20.246093 12.121094 4.695312c-5.113281-9.65625-8.808594-19.96875-10.980469-30.777343-5.292968-26.359376-.863281-54.363282 12.476563-78.851563l10.558593-19.382813 14.121094 16.960938c44.660156 53.648438 101.226563 85.472656 168.363282 94.789062-2.742188-18.902343-.6875-37.144531 6.113281-53.496093 7.917969-19.039063 22.003906-35.183594 40.722656-46.691407 20.789063-12.777343 46-18.96875 70.988281-17.433593 26.511719 1.628906 50.582032 11.5625 69.699219 28.746093 9.335937-2.425781 16.214844-5.015624 25.511719-8.515624 5.59375-2.105469 11.9375-4.496094 19.875-7.230469l29.25-10.078125-19.074219 54.476562c1.257813-.105468 2.554687-.195312 3.910156-.253906l31.234375-1.414062-18.460937 25.230468c-1.058594 1.445313-1.328125 1.855469-1.703125 2.421875-1.488282 2.242188-3.339844 5.03125-28.679688 38.867188-6.34375 8.472656-9.511718 19.507812-8.921875 31.078125 2.246094 43.96875-3.148437 83.75-16.042969 118.234375-12.195312 32.625-31.09375 60.617187-56.164062 83.199219-31.023438 27.9375-70.582031 47.066406-117.582031 56.847656-23.054688 4.796875-47.8125 7.203125-73.4375 7.203125zm0 0"
            fill="#fff9d4"
          />
        </svg>
      </a>
      <a
        aria-label="Sneakerbox on Facebook"
        href="#"
        onClick={null}
        target="_self"
      >
        <svg
          alt="facebook"
          className="facebook-logo"
          focusable="false"
          height="50px"
          onMouseEnter={[Function]}
          onMouseLeave={[Function]}
          viewBox="0 0 512 512"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="m437 0h-362c-41.351562 0-75 33.648438-75 75v362c0 41.351562 33.648438 75 75 75h151v-181h-60v-90h60v-61c0-49.628906 40.371094-90 90-90h91v90h-91v61h91l-15 90h-76v181h121c41.351562 0 75-33.648438 75-75v-362c0-41.351562-33.648438-75-75-75zm0 0"
            fill="#fff9d4"
          />
        </svg>
      </a>
      <a
        aria-label="Sneakerbox on Snapchat"
        href="#"
        onClick={null}
        target="_self"
      >
        <svg
          alt="snapchat"
          className="snapchat-icon"
          focusable="false"
          height="50px"
          onMouseEnter={[Function]}
          onMouseLeave={[Function]}
          viewBox="0 -16 511.99949 511"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="m489.4375 332.257812-22.179688-8.597656c-34.984374-13.554687-56.421874-46.179687-59.726562-83.160156h44.46875c16.570312 0 30-13.433594 30-30 0-16.570312-13.429688-30-30-30h-44.9375l1.300781-14.253906c8.0625-88.8125-62.910156-165.746094-152.363281-165.746094-88.996094 0-160.46875 76.476562-152.363281 165.746094l1.300781 14.253906h-44.9375c-16.570312 0-30 13.429688-30 30 0 16.566406 13.429688 30 30 30h45.839844l-2.6875 12.449219c-6.957032 32.257812-27.613282 58.683593-58.359375 70.679687l-20.523438 8.027344c-9.785156 2.199219-17.84375 8.671875-21.738281 17.609375-4.246094 9.753906-3.105469 21.222656 3.089844 30.671875 12.117187 18.53125 39.304687 25.328125 61.949218 29.722656 5.726563 24.988282 18.96875 36.050782 39.375 32.929688 44.707032-6.871094 47.710938-3.367188 63.828126 8.230468 7.207031 5.1875 17.183593 11.648438 32.273437 19.628907 16.007813 6.59375 34.128906 9.917969 52.425781 9.917969 20.054688 0 41.296875-3.984376 59.140625-12.027344 3.558594-2.050782 7.542969-4.730469 11.9375-7.730469 18.515625-12.601563 39.210938-28.316406 71.933594-19.734375 11.9375 3.105469 22.019531 2.109375 29.972656-2.972656 11.453125-7.292969 14.238281-20.097656 15.777344-28.34375 23.726563-4.746094 55.253906-11.527344 65.214844-35.464844 3.675781-8.863281 3.367187-18.574219-.851563-26.632812-4.234375-8.070313-12.042968-13.855469-19.160156-15.203126zm0 0"
            fill="#fff9d4"
          />
        </svg>
      </a>
      <a
        aria-label="Sneakerbox on Pinterest"
        href="#"
        onClick={null}
        target="_self"
      >
        <svg
          alt="pinterest"
          className="pinterest-icon"
          focusable="false"
          height="50px"
          onMouseEnter={[Function]}
          onMouseLeave={[Function]}
          viewBox="-62 0 512 512"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="m60.945312 278.21875c.640626-1.597656 7.050782-26.230469 7.5-27.898438-10.007812-15.058593-21.125-64.371093-12.597656-104.398437 9.199219-58.730469 71.4375-87.601563 130.199219-87.601563v-.109374c73.570313.046874 128.640625 40.980468 128.699219 111.476562.046875 55.179688-33.195313 128.117188-89.957032 128.117188-.015624 0-.027343 0-.042968 0-20.257813 0-45.90625-9.1875-52.632813-18.210938-7.761719-10.398438-9.667969-23.230469-5.566406-36.941406 10.050781-32.082032 22.867187-70.511719 24.363281-96.136719 1.386719-24.183594-15.773437-39.917969-38.027344-39.917969-16.746093 0-38.496093 9.726563-49.335937 37.058594-8.953125 22.707031-8.761719 46.480469.585937 72.671875 3.644532 10.238281-16.15625 76.984375-22.5 98.71875-15.761718 53.992187-37.339843 122.304687-32.726562 160.347656l4.453125 36.605469 22.367187-29.3125c30.953126-40.519531 62.957032-145.332031 71.484376-170.835938 25.210937 32.648438 77.710937 33.585938 83.832031 33.585938 75.183593 0 160.4375-74.65625 158.019531-178.5625-2.121094-91.121094-68.808594-166.875-188.632812-166.875v.117188c-113.976563 0-180.5 60.835937-196.785157 137.703124-14.914062 71.273438 18.253907 125.519532 57.300781 140.398438zm0 0"
            fill="#fff9d4"
          />
        </svg>
      </a>
    </div>
    <div
      className="Copyright"
    >
      <p>
        CopyrightÂ© 
        2019
         908 Productions Inc. All rights reserved.
      </p>
    </div>
  </footer>
</div>
`;
